dist/bin/scala

```diff
@@ -101,7 +114,17 @@ if [ $execute_script == true ]; then
   if [ "$CLASS_PATH" ]; then
     cp_arg="-classpath \"$CLASS_PATH\""
   fi
-  eval "\"$PROG_HOME/bin/scalac\" $cp_arg ${java_options[@]} ${residual_args[@]} -script $target_script ${script_args[@]}"
+  setScriptName="-Dscript.path=$target_script"
+  target_jar="${target_script%.*}.jar"
+  jar_found=false
+  [[ $save_compiled == true && -f "$target_jar" ]] && jar_found=true
+  if [[ $jar_found == true && "$target_jar" -nt "$target_script" ]]; then
+    java $setScriptName -jar "$target_jar" "${script_args[@]}"
+  else
+    [[ $save_compiled == true && -f $target_jar ]] && rm -f $target_jar
```

Comment:
You could replace the first operand of `&&` with `$jar_found == true`.

Reply:
I reworked that section, removing `$jar_found`:
```
  if [[ $save_compiled == true && "$target_jar" -nt "$target_script" ]]; then
```